gpflow/optimizers/scipy.py

```diff
@@ -53,6 +53,7 @@ def minimize(
         """
         if not callable(closure):
             raise TypeError("Callable object expected.")  # pragma: no cover
+        variables = tuple(variables)
```

Comment:
why convert to tuple?

Reply:
in case it IS a single-pass iterator because we need it afterwards...

Reply:
good point